### PR TITLE
Fixed the labeling issue in Gateway REST API

### PR DESCRIPTION
**Pull request by 1akshitha**
Fixes https://github.com/wso2/product-apim/issues/9025

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java [hunk 1]
```diff
@@ -71,7 +72,7 @@ public boolean deployAPI(String apiId, String gatewayLabel) throws  ArtifactSync
                 try {
                     String gatewayRuntimeArtifact = artifactRetriever.retrieveArtifact(apiId, gatewayLabel,
                             APIConstants.GatewayArtifactSynchronizer.GATEWAY_INSTRUCTION_PUBLISH);
-                    if (gatewayRuntimeArtifact != null) {
+                    if (!StringUtils.isEmpty(gatewayRuntimeArtifact)) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java:75]: StringUtils.isnotempty

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java [hunk 2]
```diff
@@ -120,19 +121,20 @@ public boolean deployAllAPIsAtGatewayStartup (Set<String> assignedGatewayLabels)
                         for (String APIruntimeArtifact :gatewayRuntimeArtifacts){
                             GatewayAPIDTO gatewayAPIDTO = null;
                             try {
-                                if (APIruntimeArtifact != null) {
+                                if (!StringUtils.isEmpty(APIruntimeArtifact)) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java:124]: fix variable naming and same goes here

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java [hunk 3]
```diff
@@ -143,9 +148,26 @@ public String retrieveArtifact(String APIId, String gatewayLabel, String gateway
...
             Map <String, String> apiAttribute = new HashMap<>();
+
             JSONObject artifactObject = new JSONObject(responseString);
-            String apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
-            String label = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABEL);
+            String label = null;
+            String apiId = null;
+            try {
+                apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
+                String labelsStr = artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABELS).toString();
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java:157]: JSONArrray

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java [hunk 4]
```diff
@@ -143,9 +148,26 @@ public String retrieveArtifact(String APIId, String gatewayLabel, String gateway
...
+
+                Set<String> labelsSet = new Gson().fromJson(labelsStr, new TypeToken<HashSet<String>>(){}.getType());
+                Set<String> gatewaySubscribedLabel = gatewayArtifactSynchronizerProperties.getGatewayLabels();
+                if (!labelsSet.isEmpty() || !gatewaySubscribedLabel.isEmpty()){
+                    labelsSet.retainAll(gatewaySubscribedLabel);
+                    if (!labelsSet.isEmpty()){
+                        label = labelsSet.iterator().next();
+                    }
+                }
+            } catch (Exception e ){
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java:167]: can't we catch specific exception

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java (hunk 3 above)
Comment (1akshitha) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java:157]: Converting the JSONArray to set to get the intercept of two sets.

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java (hunk 1 above)
Comment (1akshitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java]: Fixed f1fefa5ebc85177015c7bdac17899ebac013acdd

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java (hunk 2 above)
Comment (1akshitha) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java]: Fixed f1fefa5ebc85177015c7bdac17899ebac013acdd

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java (hunk 4 above)
Comment (1akshitha) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java]: Fixed f1fefa5ebc85177015c7bdac17899ebac013acdd

**Pull request merged**